fusedrug_examples/interaction/drug_target/affinity_prediction/t5_mol/tokenizer/multi_tokenizer.py

```diff
@@ -0,0 +1,1304 @@
+        """
+        # super.__init__()
+        # TODO: if there is only one tokenizer, leave it as is, without changing its id mappings. Not needed - if there's only one, we can just load its json using load_from_jsons
+        self.tokenizers_info = ModularTokenizer.cfg_list_2_dict(tokenizers_info)
+
+        if not load_adjusted_jsons:
+            # store special tokens in a list to preserve their order:
+            all_special_tokens: List = list([])
+
+            # collect all special tokens (without keeping indices):
+            for t_type in self.tokenizers_info:
+                t_info = self.tokenizers_info[t_type]
+                t_json = json.load(open(t_info["json_path"]))
+                self.tokenizers_info[t_type]["json_instance"] = t_json
+                model_type = t_json["model"]["type"]
+                assert (
+                    model_type == "BPE"
+                ), f"Only tokenizer models of type BPE are supported (got {model_type}). Other types were not tested. Comment this assertion at your own risk"
+
+                part_special_tokens = ModularTokenizer.get_special_tokens(t_json, force_special=False)
+                part_special_tokens = [t for t in part_special_tokens if t not in all_special_tokens]
+                all_special_tokens = all_special_tokens + part_special_tokens
+
+            all_special_token_structs = ModularTokenizer.build_special_token_list(all_special_tokens)
+            # rearrange regular token indices
+            next_index = max([t["id"] for t in all_special_token_structs]) + 1
+        else:
+            for t_type in self.tokenizers_info:
+                t_info = self.tokenizers_info[t_type]
+                t_json = json.load(open(t_info["modular_json_path"]))
+                self.tokenizers_info[t_type]["json_instance"] = t_json
+
+        for t_type in self.tokenizers_info:
+            t_info = self.tokenizers_info[t_type]
+            t_json = self.tokenizers_info[t_type]["json_instance"]
+            # operations on the tokenizer json
+            if not load_adjusted_jsons:
+                t_json["added_tokens"] = all_special_token_structs
+                (t_json["model"]["vocab"], next_index,) = ModularTokenizer.remap_vocab(
+                    vocab=t_json["model"]["vocab"],
+                    special_token_structs=all_special_token_structs,
+                    starting_index=next_index,
+                )
+            # end operations on json
+            json_str = json.dumps(t_json)
+            tokenizer_inst = Tokenizer.from_str(json_str)
+            if "max_len" in t_info and t_info["max_len"] is not None:
+                max_size = t_info["max_len"]
+                tokenizer_inst.enable_truncation(
+                    max_length=max_size,
+                    direction="right",
+                )
+            json_str = tokenizer_inst.to_str()
```

Comment:
is this the instance we are going to save for load, would it include also the max_len?
I think that max_len, should be specified explicitly and not be stored in the json file.

Reply:
It will not include max_len - it's not a part of the tokenizer. A tokenizer only contains the mapping, it shouldn't be aware of the limitations of use, and a single tokenizer can be instantiated to have different max_len values. max_len is taken from the config during load, or it can be passed to the encode method.